### ISSUE TITLE
User already registered error

### ISSUE THREAD

**Issue by ganeshnegi** [processone/ejabberd#2180]
> I am using ejabberd 16.01

> operating system (version) : Ubuntu 16.04 LTS

**I have followed the below steps to install ejabberd:**

sudo apt-get install ejabberd

I created a database (testdb) in Postgres.
Created a user in Postgres and migrated the schema as psql testdb < /path/to/my/pg.sql
(I downloaded pg.sql from https://github.com/processone/ejabberd/blob/master/sql/pg.sql)

After this, I installed the erlang postgres driver from https://github.com/processone/p1_pgsql.git

It was working fine until using the internal Mnesia database. Now I want to use PostgreSQL to store the ejabberd messages.

**configuration changes I made in ejabberd.yml are as:**
```yaml
auth_method: odbc

hosts:
  - "localhost"

##
## PostgreSQL server:
## 
odbc_type: pgsql
odbc_server: "127.0.0.1"
odbc_database: "testdb"
odbc_username: "sqldev"
odbc_password: "sqlpass"

##
## Modules enabled in all ejabberd virtual hosts.
##
modules:
  mod_muc:
    ## host: "conference.@HOST@"
    db_type: odbc
    access: muc
    access_create: muc_create
    access_persistent: muc_create
    access_admin: muc_admin
  
  mod_offline:
    db_type: odbc
    access_max_user_messages: max_user_offline_messages
  
  mod_mam:
    db_type: odbc
    default: always
```

_When I run ejabberdctl register admin localhost password, I get the following message:_
User admin@localhost already registered at node 'ejabberd@ganesh'

When I check in the database, the users table is empty.

Please suggest what I am missing and how to solve this issue.

Thank you very much.

**Comment by zinid** [comment]
> I am using ejabberd 16.01

This is an extremely old version, ejabberd is evolving rapidly, two years for it is abyss.

> db_type: odbc

`db_type: sql`

> User admin@localhost already registered at node 'ejabberd@ganesh'

It's not clear, did you add this "localhost" as a new hostname to the existing ones? If yes, and you have several virtual hosts, do you have a separate pgsql database for each?

**Comment by zinid** [comment]
> After this, I installed the erlang postgres driver from https://github.com/processone/p1_pgsql.git

You should never do this. You should compile ejabberd with postgresql support:
```
$ ./configure --enable-pgsql ...
```

**Comment by cromain** [comment]
Also, you are applying pg.sql schema from HEAD of master (https://github.com/processone/ejabberd/blob/master/sql/pg.sql) instead of the one included in your 16.01 package => This will never works.

You'd better use our latest .deb available here: https://www.process-one.net/downloads/downloads-action.php?file=/ejabberd/17.11/ejabberd_17.11-0_amd64.deb
